src/blockchain/electrum.rs

```diff
@@ -108,7 +108,10 @@ impl WalletSync for ElectrumBlockchain {
         let mut block_times = HashMap::<u32, u32>::new();
         let mut txid_to_height = HashMap::<Txid, u32>::new();
         let mut tx_cache = TxCache::new(database, &self.client);
-        let chunk_size = self.stop_gap;
+
+        // Ensure chunk_size > 0, otherwise we will result in an infinite loop
```

Comment:
I understand better what is going on now. We must find a gap greater than `stop_gap` in order to stop.
So I think this comment should be: 

> set the chunk_size to the smallest value that we can find a gap greater than stop gap.

It is also a performance issue to set it to `stop_gap` since you will do two requests when you should only do one on any empty wallet (for example).

Reply:
Changed in 475b005eb9151284885600896e8f0f50a45cd1b2.